scarlet-websocket-okhttp/src/main/java/com/tinder/scarlet/websocket/okhttp/OkHttpClientUtils.kt

```diff
@@ -6,15 +6,34 @@
 
 package com.tinder.scarlet.websocket.okhttp
 
+import android.os.Build
+import android.security.NetworkSecurityPolicy
 import com.tinder.scarlet.WebSocket
 import com.tinder.scarlet.websocket.okhttp.request.RequestFactory
 import com.tinder.scarlet.websocket.okhttp.request.StaticUrlRequestFactory
+import okhttp3.HttpUrl
 import okhttp3.OkHttpClient
+import java.net.UnknownServiceException
 
 fun OkHttpClient.newWebSocketFactory(requestFactory: RequestFactory): WebSocket.Factory {
     return OkHttpWebSocket.Factory(OkHttpClientWebSocketConnectionEstablisher(this, requestFactory))
 }
 
 fun OkHttpClient.newWebSocketFactory(url: String): WebSocket.Factory {
+    if (url.startsWith("ws:", ignoreCase = true)) {
+        try {
+            if ((Build.VERSION.SDK_INT > 23 &&
```

Comment:
I like the [`Platform#isCleartextTrafficPermitted`](https://github.com/square/okhttp/blob/master/okhttp/src/jvmMain/kotlin/okhttp3/internal/platform/Platform.kt#L136) abstraction. we could tackle it in the next PR